src/streamlit_app/app.py

```diff
@@ -37,20 +40,24 @@
 logging.basicConfig(level=logging.INFO)
 logger.setLevel(logging.INFO)
 
-# ------------------------- ENVIRONMENT VARIABLES -----------------------------
-
-
-# ------------------------------ MAIN FUNCTIONS -------------------------------
-
 
 def main():
     """
     Main function for the Streamlit Application
     """
+    data_src = build_page()
+
+    url = get_media_url(data_src)
+
+    handle_analysis(url)
```

Comment:
I like how this is structured.

Reply:
Thanks! I think more could be done to organize things a bit more but based on what we had this worked pretty well. One of my favorite things is extracting methods like this. 😃